src/main/java/com/tcg/lwjgllearning/graphics/g3d/mesh/AbstractMesh.java

```diff
@@ -0,0 +1,39 @@
+package com.tcg.lwjgllearning.graphics.g3d.mesh;
+
+import java.util.HashMap;
+
+import com.tcg.lwjgllearning.graphics.Drawable;
+import com.tcg.lwjgllearning.math.Quaternion;
+import com.tcg.lwjgllearning.math.Transform3D;
+import com.tcg.lwjgllearning.math.Vector3;
+import com.tcg.lwjgllearning.utils.Disposable;
+
+import static org.lwjgl.opengl.GL15.*;
+import static org.lwjgl.opengl.GL20.*;
+
+public abstract class AbstractMesh extends Transform3D implements Disposable, Drawable {
+    
+    protected HashMap<AbstractAttribute, Integer> attributeBufferMap;
```

Comment:
I can see where this is populated, but where is it used?

Reply:
It is used in the "unused" methods discussed in your first comment above. Its purpose is to allow other classes to specify an attribute in said methods without having access to the vbo identifier.